src/main/java/org/jenkinsci/plugins/mesos/MesosSlaveInfo.java

```diff
@@ -180,13 +184,27 @@ public String getImage() {
     private final String type;
     private final String dockerImage;
     private final List<Volume> volumes;
+    private final String networking;
+    private final List<PortMapping> portMappings;
 
     @DataBoundConstructor
-    public ContainerInfo(String type, String dockerImage, List<Volume> volumes)
+    public ContainerInfo(String type, String dockerImage, List<Volume> volumes, String networking, List<PortMapping> portMappings)
       throws FormException {
       this.type = type;
       this.dockerImage = dockerImage;
       this.volumes = volumes;
+
+      if (networking == null) {
+          this.networking = Network.BRIDGE.toString();
+      } else {
+          this.networking = networking;
+      }
+
+      if (Network.HOST.equals(Network.valueOf(networking))) {
```

Comment:
Should a PortMapping perhaps set the host port if the networking is HOST as the mapping is really just implied in Docker?